src/main/java/com/fasterxml/jackson/databind/deser/DeserializerCache.java

```diff
@@ -57,6 +57,21 @@ public DeserializerCache() {
     public DeserializerCache(int maxSize) {
         int initial = Math.min(64, maxSize>>2);
         _cachedDeserializers = new LRUMap<>(initial, maxSize);
+        _incompleteDeserializers = new HashMap<JavaType, JsonDeserializer<Object>>(8);
+    }
+
+    public DeserializerCache(LookupCache<JavaType, JsonDeserializer<Object>> cachedDeserializers,
```

Comment:
Should not be public, I think, only called from `withCache()`. Can also drop `incompleteDeserializers` for reasons mentioned above.